### PR TITLE
Introduce a centralized library to call the Checkmk API

### PR DESCRIPTION
**Pull request by godspeed-you**
The code to contact the Checkmk API is redundantly placed all over the modules. This projects aims to centralize this code into a single library to make our lifes more easy. Later changes may be done in one single place instead of changing several dozens of code lines.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe):

## What is the current behavior?

- Currently, if the API headers or other static information is changed, this needs to be adapted in an unclear amount of files and code lines. 
- Everytime a new API call to Checkmk is added, bugs or slightly different behaviour may be introduced.

## What is the new behavior?

- The centralized file may be called by every module to have a defined set of functionality returned information.
- This set may be used to derive a module specific class to address specific requirements.
- The result of this API call is stored in a defined and clearly structured data set. This allows to automatically provide a clear and reliable set of information after the API call.

## Other information

This PR contains a migrated module as an example on how to use this new library. The changes have been reduced to an absolute minimum to minimize potential bugs. So potential further refactorings may be skipped to have this change less complex.

### CODE REVIEW

**Review by lgetwan — COMMENTED** [review]
I really like the way you unified and simplified the API calls!
I have a question regarding the definition of the inheriting class, though. In some modules, there are multiple, different POST or GET calls. How would you define the related 
`def post(self)`
calls for those?

**Comment by msekania** [comment]
related #176

**Comment by msekania** [comment]
@godspeed-you,

First of all thanks,

Concerning module, in essence it should at least cover all already implemented usage in different modules, otherwise it will be yet another modules which one has to adjust manually per module and not the central one.

**Comment by lgetwan** [comment]
I agree to update the modules one by one. And I agree, it's probably easier to have separate feature branches for each module change.
Thanks for updating the user module, as well!

**Comment by msekania** [comment]
@godspeed-you,

when structure is fixed and accepted, I will also help with group modules.

File: plugins/modules/activation.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 # If this is part of a collection, you need to use semantic versioning,
 # i.e. the version is of the form "2.5.0" and not "2.4".
-version_added: "0.0.1"
+version_added: "0.0.2"
```
Comment (robin-tribe29) [inline plugins/modules/activation.py:18]: Why did you change this? This reflects the version of the collection the module was introduced to. There is no independent versioning of modules.

**Review by robin-tribe29 — CHANGES_REQUESTED** [review]

File: plugins/modules/activation.py (hunk 1 above)
Comment (godspeed-you) [reply plugins/modules/activation.py]: My bad. I overlooked the name of the variable and only read the comment. This is resolved by the last commit.

**Review by lgetwan — APPROVED** [review]
Looks fine!

**Review by robin-tribe29 — APPROVED** [review]

**Pull request merged**